src/commands/generate/models.ts

```diff
@@ -16,31 +16,68 @@ export default class Models extends Command {
   static description = 'Generates typed models';
 
   static args = [
-    { 
-      name: 'language', 
-      description: 'The language you want the typed models generated for.', 
-      options: Object.keys(Languages), 
-      required: true 
+    {
+      name: 'language',
+      description: 'The language you want the typed models generated for.',
+      options: Object.keys(Languages),
+      required: true
     },
     { name: 'file', description: 'Path or URL to the AsyncAPI document, or context-name', required: true },
   ];
 
   static flags = {
     help: Flags.help({ char: 'h' }),
-    output: Flags.string({ char: 'o', description: 'The output directory where the models should be written to. Omitting this flag will write the models to `stdout`.', required: false}),
+    output: Flags.string({
+      char: 'o',
+      description: 'The output directory where the models should be written to. Omitting this flag will write the models to `stdout`.',
+      required: false
+    }),
+    /**
+     * TypeScript specific options
+     */
+    modelType: Flags.string({
+      type: 'option',
+      options: ['class', 'interface'],
+      description: 'TypeScript specific, define which type of model needs to be generated.',
+      required: false,
+    }),
+    enumType: Flags.string({
+      type: 'option',
+      options: ['enum', 'union'],
+      description: 'TypeScript specific, define which type of enums needs to be generated.',
+      required: false,
+    }),
+    moduleSystem: Flags.string({
+      type: 'option',
+      options: ['ESM', 'CJS'],
+      description: 'TypeScript specific, define the module system to be used.',
+      required: false,
+    }),
+    exportType: Flags.string({
+      type: 'option',
+      options: ['default', 'named'],
+      description: 'TypeScript specific, define which type of export needs to be generated.',
+      required: false,
+    }),
```

Comment:
I don't think so that this solution is good for future features. At the moment we support 5-6 languages and in future we can support many more and hardcode some flags for particular language is a bad solution. I think that we should go with idea like having `--config` which should be stringified JSON and the shape of the`--config` should be determine by chosen language. What do you think?

cc @jonaslagoni

Reply:
And value for the `--config` flag should meet the interface of options that can be passed to a given generator - e.g. for TS - https://github.com/asyncapi/modelina/blob/next/src/generators/typescript/TypeScriptGenerator.ts#L17 - of course without options where you need to pass some runtime references.

Reply:
I actually think we should have both 🤔

Cause some would much rather just use pure parameters, especially if they need to quickly generate the models, then they dont care about the config file.

Reply:
But you can have conflicts with parameters names for other languages, that's my whole point 😄 Also if we go with `--config` we don't need to worry about updating CLI when new Modelina will release, because we will pass options shape.

Reply:
If there are conflicts, we can easily adapt the parameters 🙂 I vote to have the parameters and in future major versions you can always remove them if you find them too challenging, when the config feature is added 😄 

> Also if we go with --config we don't need to worry about updating CLI when new Modelina will release, because we will pass options shape.

That is true yes.

But the configuration feature is not likely to be implemented this year.

Reply:
So I opt for having prefix in flags like `--tsExportType`

Reply:
Updated flag names with a ts prefix.

Reply:
Do you want to implement a `--config` option as part of this PR or as part of a future enhancement?

Reply:
Definitely future 🙂